### PR TITLE
rcutorture: optimise code using avocado modules

### PR DESCRIPTION
**Pull request by abdhaleegit**
Fixed code issues and reused linux_modules to optimise code

Signed-off-by: Abdul Haleem <abdhalee@linux.vnet.ibm.com>

### CODE REVIEW

File: generic/rcutorture.py [hunk 1]
```diff
@@ -39,16 +40,13 @@ def setUp(self):
         """
         self.results = []
         self.log.info("Check if CONFIG_RCU_TORTURE_TEST is enabled\n")
-        ret = os.system(
-            'cat /boot/config* | grep CONFIG_RCU_TORTURE_TEST=m')
-        if ret != 0:
+        ret = linux_modules.check_kernel_config('CONFIG_RCU_TORTURE_TEST')
+        if ret == 0:
```
Comment (PraveenPenguin) [inline generic/rcutorture.py]: can `if  ret == linux_modules.MODULE` use instead of `if ret == 0` as  linux_modules having a MODULE global variable .

Comment (abdhaleegit) [inline generic/rcutorture.py]: the module can return 0/1/2, but what I need is '0', so it should be ;
if linux_modules.check_kernel_config('CONFIG_RCU_TORTURE_TEST') == 0:

Comment (PraveenPenguin) [inline generic/rcutorture.py]: @abdhaleegit  yes ` linux_module.check_kernel_config` return 0 , 1 ,2 but i think in this API we have a global variable as `NOT_SET,MODULE,BUILTIN `respectively `0,1,2 `

oh sorry  we can use` linux_modules.NOT_SET`

**Comment by PraveenPenguin** [comment]
@abdhaleegit  ping

File: generic/rcutorture.py (hunk 1 above)
Comment (abdhaleegit) [inline generic/rcutorture.py]: I will be sending updated code today

Comment (abdhaleegit) [inline generic/rcutorture.py]: With changes from the last pull request, I think we are good to go, Please push the changes

Comment (PraveenPenguin) [inline generic/rcutorture.py]: @abdhaleegit thanks  will push

**Comment by PraveenPenguin** [comment]
@abdhaleegit please attach job log in this PR 

Thanks

**Comment by PraveenPenguin** [comment]
@abdhaleegit facing some issue while run can you please help to figure out this

```
[stderr] For more details see dmesg(1).

Reproduced traceback from: /usr/local/lib/python2.7/dist-packages/avocado_framework-40.0-py2.7.egg/avocado/core/test.py:451
Traceback (most recent call last):
  File "/root/rcutorture.py", line 107, in test
    res = process.system_output('dmesg | grep "rcu-torture: Reader"')
  File "/usr/local/lib/python2.7/dist-packages/avocado_framework-40.0-py2.7.egg/avocado/utils/process.py", line 1139, in system_output
    sudo=sudo)
  File "/usr/local/lib/python2.7/dist-packages/avocado_framework-40.0-py2.7.egg/avocado/utils/process.py", line 1046, in run
    raise CmdError(cmd, sp.result)
CmdError: Command 'dmesg | grep "rcu-torture: Reader"' failed (rc=1)

Traceback (most recent call last):

  File "/usr/local/lib/python2.7/dist-packages/avocado_framework-40.0-py2.7.egg/avocado/core/test.py", line 506, in _run_avocado
    raise test_exception

CmdError: Command 'dmesg | grep "rcu-torture: Reader"' failed (rc=1)

ERROR 1-rcutorture.py:Rcutorture.test -> CmdError: Command 'dmesg | grep "rcu-torture: Reader"' failed (rc=1)

Not logging /sys/devices/system/cpu/cpu0/cpufreq/scaling_governor (file does not exist)
Not logging /proc/pci (file does not exist)
Test results available in /root/avocado/job-results/job-2016-09-14T00.46-f066333
root@ubuntu:~# 
```

**Comment by PraveenPenguin** [comment]
@abdhaleegit ping

**Pull request merged**